### ISSUE TITLE
How to derive depth maps from point maps and camera parameters ?

### ISSUE THREAD

**Issue by stanley-313** [facebookresearch/vggt#222]
Dear author,

In the paper, you mention that the point map derived from the depth map and camera pose has higher accuracy. I'm curious about how the depth accuracy compares between the depth branch and the point cloud + camera pose branch. However, the results I obtained seem incorrect. Here is my code:

Did I miss something or make a mistake?

```
def point_2_depth(points, intrinsic, extrinsic):
    ''' Convert world points to depth maps.
    Args:
        points (torch.Tensor): (N, H, W, 3)
        intrinsic (torch.Tensor): (N, 3, 3)
        extrinsic (torch.Tensor): (N, 3, 4)
    Returns:
        depth (torch.Tensor): (N, H, W)
    '''

    N, H, W, _ = points.shape

    points_homo = torch.cat([points, torch.ones_like(points[..., :1])], dim=-1)

    camera_points = torch.einsum('nij,nhwj->nhwi', extrinsic, points_homo)  # (N, H, W, 3)

    depth = camera_points[..., 2]  # (N, H, W)

    return depth
```

**Comment by jytime** [comment]
Hi can you have a look at this?

https://github.com/facebookresearch/vggt/blob/f4ceb3671948fd8e5e6f5e7fddf0b1b9bacd0d77/vggt/dependency/projection.py
